src/Databases/DatabaseOnDisk.cpp

```diff
@@ -760,6 +760,7 @@ ASTPtr DatabaseOnDisk::parseQueryFromMetadata(
     }
 
     ReadSettings read_settings = getReadSettings();
+    read_settings.local_fs_method = LocalFSReadMethod::read;
```

Comment:
May be use `pread` as it is the default non-async method? https://github.com/ClickHouse/ClickHouse/blob/204fd5b54317ad310559805e6ebd7225cf014975/src/IO/ReadSettings.h#L26

Reply:
With `pread`, it still uses `ReadBufferFromFilePReadWithDescriptorsCache`. So the data might not update-to-date. I think it's safer to use `read` for reading files with db_disk.